src/main/java/io/cdap/plugin/gcp/datastore/source/DatastoreSourceConfig.java

```diff
@@ -564,6 +564,8 @@ boolean shouldConnect() {
       !(containsMacro(DatastoreSourceConfig.NAME_SERVICE_ACCOUNT_FILE_PATH) ||
         containsMacro(DatastoreSourceConfig.NAME_SERVICE_ACCOUNT_JSON)) &&
       !containsMacro(DatastoreSourceConfig.NAME_PROJECT) &&
+      !(containsMacro(DatastoreSourceConstants.PROPERTY_KEY_TYPE) ||
```

Comment:
unless I'm misunderstanding the doc, shouldn't both of these required to be non-macro instead of just one?

Reply:
@albertshau I'm not sure I understand your question.

I think a customer might have a scenario where they set their Key Type to either `none` or `key literal` as needed. Given that the `key alias` is ignored when Key Type is set to none, this is a valid scenario.

The other case is where a customer always sets their Key Type as `key literal` but wants to use a Macro for the Key Alias.

Reply:
I got confused by the parantheses. This is logically equivalent to `!containsMacro(key) && !containsMacro(key_alias)`, which would be a simpler way to express it.

Same is true for the service account properties that are checked above.

Reply:
I wanted to keep it "grouped" in the same way the NAME_SERVICE_ACCOUNT_FILE_PATH and NAME_SERVICE_ACCOUNT_JSON are. I can transform `!(a || b)` into `!a && !b` if you prefer.

Reply:
yes, can you change them both. We shouldn't have a mix of them

Reply:
Done.